src/main/java/frc/robot/commands/drive/DriveFieldOrientedHeadingSnapping.java

```diff
@@ -27,11 +28,38 @@
 public class DriveFieldOrientedHeadingSnapping extends Command {
   private Drive drive;
   private XboxController controller;
+  private final Supplier<Double> xSupplier, ySupplier, zSupplier;
+  private final Supplier<Boolean> upSupplier, downSupplier, leftSupplier, rightSupplier;
 
-  /** Creates a new DriveFieldOrientedHeadingSnapping. */
-  public DriveFieldOrientedHeadingSnapping() {
+  /**
+   * Drives the robot field oriented with heading snapping. All values are suppliers to make the
+   * command more versatile.
+   *
+   * @param xSupplier The x speed
+   * @param ySupplier The y speed
+   * @param zSupplier The z speed
+   * @param upSupplier If the robot should go up
+   * @param downSupplier If the robot should go down
+   * @param leftSupplier If the robot should go left
+   * @param rightSupplier If the robot should go right
```

Comment:
"If true, the robot heading will be set to move towards (the opposing alliance wall, our alliance wall, left, or right)"